pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -140,7 +140,7 @@ public CompletableFuture<Boolean> canConsumeAsync(TopicName topicName, String ro
                 }).exceptionally(ex -> {
                     log.warn("Client with Role - {} failed to get permissions for topic - {}. {}", role, topicName,
                             ex.getMessage());
-                    return null;
+                    throw FutureUtil.wrapToCompletionException(ex);
```

Comment:
Is the logging just extra noise if the exception gets rethrown?  Could we remove the `.exceptionally` completely?

I guess `return null;` is wrong and could be `return false;` if the desire is to just log and continue by returning `false`.

Reply:
We already log the "error" multiple times. We log it in this class as a warning. We also have these logs for handling the result of this method:

https://github.com/apache/pulsar/blob/952187279227fbf46af0f68d31ede1f5567e8075/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java#L888-L902

After this change, we could have 3 lines and a stack trace. I think that is too many. I would prefer to simplify the code path and only log a single failure/error/rejection once.